python/pyarrow/types.pxi

```diff
@@ -1474,3 +1474,15 @@ def from_numpy_dtype(object dtype):
         check_status(NumPyDtypeToArrow(dtype, &c_type))
 
     return pyarrow_wrap_data_type(c_type)
+
+
+def is_boolean_object(object obj):
```

Comment:
The naming and/or placement in types module feels a bit odd.

```python
pa.types.is_integer_object(pa.types.int16())  # False
```

Perhaps `is_<type>_value`? Or how about creating a `pyarrow.inference` module with `is_bool`, `is_int`, `is_float` functions?

Reply:
I like the _value name ok